id: PYSEC-0000-CVE-2024-3573
severity:
- type: CVSS_V3
  score: CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:C/C:H/I:L/A:N
details: mlflow/mlflow is vulnerable to Local File Inclusion (LFI) due to improper
  parsing of URIs, allowing attackers to bypass checks and read arbitrary files on
  the system. The issue arises from the 'is_local_uri' function's failure to properly
  handle URIs with empty or 'file' schemes, leading to the misclassification of URIs
  as non-local. Attackers can exploit this by crafting malicious model versions with
  specially crafted 'source' parameters, enabling the reading of sensitive files within
  at least two directory levels from the server's root.
affected:
- package:
    name: mlflow
    ecosystem: PyPI
    purl: pkg:pypi/mlflow
  ranges:
  - type: GIT
    repo: https://github.com/mlflow/mlflow
    events:
    - introduced: "0"
    - fixed: 438a450714a3ca06285eeea34bdc6cf79d7f6cbc
    - fixed: 438a450714a3ca06285eeea34bdc6cf79d7f6cbc
  - type: ECOSYSTEM
    events:
    - introduced: "0"
    - fixed: 2.10.0
references:
- type: EVIDENCE
  url: https://huntr.com/bounties/8ea058a7-4ef8-4baf-9198-bc0147fc543c
- type: FIX
  url: https://github.com/mlflow/mlflow/commit/438a450714a3ca06285eeea34bdc6cf79d7f6cbc
- type: REPORT
  url: https://huntr.com/bounties/8ea058a7-4ef8-4baf-9198-bc0147fc543c
- type: WEB
  url: https://huntr.com/bounties/8ea058a7-4ef8-4baf-9198-bc0147fc543c
aliases:
- CVE-2024-3573
modified: "2025-02-03T15:50:28Z"
published: "2024-04-16T00:15:12Z"
